File: cpp/37d_1472A_CardsForFriends.cpp
#include<bits/stdc++.h>

int main(){
    int t, w, h, n, cnt;
    std::vector<std::string> ans;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>w>>h>>n;
        cnt = 1;
        while (cnt < n && (w % 2 == 0 || h % 2 == 0)){
            cnt *= 2;
            if (w % 2 == 0){
                w /= 2;
            }
            else if (h % 2 == 0){
                h /= 2;
            }
        }
        if (cnt >= n){
            ans.push_back("YES");
        }
        else{
            ans.push_back("NO");
        }
    }
    for (auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}

File: python/37d_1472A_CardsForFriends.py
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        w, h, n = map(int, input().split())
        cnt = 1
        while cnt < n and (w % 2 == 0 or h % 2 == 0):
            cnt *= 2
            if w % 2 == 0:
                w //= 2
            else:
                h //= 2
        if cnt >= n:
            ans.append("YES")
        else:
            ans.append("NO")
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()